src/it/java/com/sun/checkstyle/test/base/AbstractSunModuleTestSupport.java

```diff
@@ -0,0 +1,155 @@
+
+import com.puppycrawl.tools.checkstyle.ConfigurationLoader;
+import com.puppycrawl.tools.checkstyle.DefaultConfiguration;
+import com.puppycrawl.tools.checkstyle.PropertiesExpander;
+import com.puppycrawl.tools.checkstyle.api.CheckstyleException;
+import com.puppycrawl.tools.checkstyle.api.Configuration;
+import com.puppycrawl.tools.checkstyle.internal.utils.CheckUtil;
+import com.puppycrawl.tools.checkstyle.utils.ModuleReflectionUtil;
+
+public abstract class AbstractSunModuleTestSupport extends AbstractItModuleTestSupport {
+
+    private static final String XML_NAME = "/sun_checks.xml";
+
+    private static final Configuration CONFIGURATION;
+
+    private static final Set<Class<?>> CHECKSTYLE_MODULES;
+
+    static {
+        try {
+            CONFIGURATION = ConfigurationLoader.loadConfiguration(XML_NAME,
+                    new PropertiesExpander(System.getProperties()));
+        }
+        catch (CheckstyleException ex) {
+            throw new IllegalStateException(ex);
+        }
+        try {
+            CHECKSTYLE_MODULES = CheckUtil.getCheckstyleModules();
+        }
+        catch (IOException ex) {
+            throw new IllegalStateException(ex);
+        }
+    }
+
+    @Override
+    protected ModuleCreationOption findModuleCreationOption(String moduleName) {
+        ModuleCreationOption moduleCreationOption = ModuleCreationOption.IN_CHECKER;
+
+        for (Class<?> moduleClass : CHECKSTYLE_MODULES) {
+            if (moduleClass.getSimpleName().equals(moduleName)
+                    || moduleClass.getSimpleName().equals(moduleName + "Check")) {
+                if (ModuleReflectionUtil.isCheckstyleTreeWalkerCheck(moduleClass)
+                        || ModuleReflectionUtil.isTreeWalkerFilterModule(moduleClass)) {
+                    moduleCreationOption = ModuleCreationOption.IN_TREEWALKER;
+                }
+                break;
+            }
+        }
+
+        return moduleCreationOption;
+    }
+
+    @Override
+    protected DefaultConfiguration createModuleConfig(Class<?> clazz) {
+        return new DefaultConfiguration(clazz.getName());
+    }
+
+    /**
+     * Returns {@link Configuration} instance for the given module name.
+     * This implementation uses {@link #getModuleConfig(String, String)} method inside.
+     * @param moduleName module name.
+     * @return {@link Configuration} instance for the given module name.
+     */
+    protected static Configuration getModuleConfig(String moduleName) {
+        return getModuleConfig(moduleName, null);
+    }
+
+    /**
+     * Returns {@link Configuration} instance for the given module name.
+     * This implementation uses {@link #getModuleConfig(String)} method inside.
+     * @param moduleName module name.
+     * @param moduleId module id.
+     * @return {@link Configuration} instance for the given module name.
+     */
+    protected static Configuration getModuleConfig(String moduleName, String moduleId) {
```

Comment:
You copied this method twice. Now there is duplicate code in `AbstractGoogleModuleTestSupport` and `AbstractSunModuleTestSupport`.
Does it make sense to move this into a separate class, common to Google and Sun tests?
It seems that only `XML_NAME` can be left from `Abstract{Google,Sun}ModuleTestSupport`.

Reply:
This is not the only method copied between google and sun. The only difference between them is the config location, everything else is a copy/paste.
It is also not the only copy in the entire test structure. `AbstractItModuleTestSupport` and `AbstractModuleTestSupport` are basically copies but the main difference between them is the warning system required in the input files.

Reply:
I can't split them apart because this method is `static` and the fields are `static final` and all these methods use these fields. This is why they were duplicated in the first place.